apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PooledSession.cs

```diff
@@ -270,9 +282,16 @@ public async Task<CommitResponse> CommitAsync(CommitRequest request, CallSetting
         {
             CheckNotDisposed();
             GaxPreconditions.CheckNotNull(request, nameof(request));
-            GaxPreconditions.CheckState(TransactionId != null, "Cannot commit a PooledSession with no associated transaction");
+            var transactionId = TransactionId;
+            if (transactionId == null)
```

Comment:
And given that that involves making a request, it can't just be a matter of taking a lock. I think we need to take a step back and work out the situations in which this can reasonably be called multiple times from different threads. We could potentially prevent that from happening (safely remember whether commit has been called, and throw an exception if it has) which would at least make things easier. But I don't know whether that's reasonable.